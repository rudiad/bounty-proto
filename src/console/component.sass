.console
  bottom: 0
  position: absolute
  border: 1px solid black
  padding: .7em
  left: 0
  right: 0
  margin-left: auto
  margin-right: auto
  height: 10em
  color: white
  background-color: #241d30
  opacity: 0.8
  overflow-y: auto
  font-family: "Lucida Console", Monaco, monospace
  z-index: -1
