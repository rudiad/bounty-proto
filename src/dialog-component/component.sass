$top-bar-dimension: 40px

.dialog__backdrop
  position: fixed
  top: 0
  bottom: 0
  left: 0
  right: 0
  background-color: #19171C
  opacity: 0.75
  padding: 50
  height: 100vh

.dialog
  background-color: white
  border-radius: 5
  width: 66%
  margin: 0 auto
  border: 1px solid #DAD8DE
  position: absolute
  top: 20%
  left: 20%

.container
  display: flex

.container__block
  display: inline-block

.container__left-block
  @extend .container__block
  float: left
  margin-left: 20px

.container__right-block
  @extend .container__block
  float: right

.top-bar
  @extend .container
  height: $top-bar-dimension
  justify-content: space-between

.top-bar-element
  width: auto
  text-align: center
  vertical-align: middle
  line-height: $top-bar-dimension
  font-weight: lighter

.top-bar-element-left
  @extend .container__left-block
  @extend .top-bar-element

.top-bar-element-right
  @extend .container__right-block
  @extend .top-bar-element

.dialog__divider-width
  width: 96%

@mixin bottom-bar-box($color, $bg-color)
  color: $color
  background-color: $bg-color
  border: 1px solid #7D55C7
  float: left
  text-align: center
  width: 80px
  height: 30px
  vertical-align: middle
  line-height: 30px
  cursor: pointer
  font-size: 16px
  margin-left: 20px
  margin-top: 10px
  margin-bottom: 30px

.title
  font-size: 13px
  font-weight: lighter
