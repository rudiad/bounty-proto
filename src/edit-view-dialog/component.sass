@import "../dialog-component/component.sass"

.edit-view__background
  @extend .dialog__backdrop

.edit-view__dialog-component
  @extend .dialog
  height: 275px
  width: 275px
  left: 40%

.edit-view__dialog-mobile
  @extend .dialog
  height: 275px
  width: 415px
  left: 35%

.edit-view__dialog-overlay
  @extend .dialog
  height: 275px
  width: 275px
  left: 40%

.edit-view__content
  @extend .container__left-block
  overflow-y: auto
  height: 50%
  width: 95%
  display: flex
  flex-direction: column

.edit-subcontainer__input
  margin-top: .5em
  margin-bottom: .5em

.dialog__top-bar-container
  @extend .top-bar

.top-bar-container__title
  @extend .top-bar-element-left
  margin-top: 5px

.top-bar-container__escape
  @extend .top-bar-element-right
  width: $top-bar-dimension
  cursor: pointer
  width: 4em
  margin-top: 5px

.dialog__divider
  @extend .dialog__divider-width

.dialog__bottom-bar
  @extend .container__left-block

.bottom-bar__save
  @include bottom-bar-box(white, #7D55C7)

.bottom-bar__cancel
  @include bottom-bar-box(#7D55C7, white)
  margin-left: 10px
