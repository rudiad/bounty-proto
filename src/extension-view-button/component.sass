.circle-button
  border-radius: 50%
  width: 48px
  height: 48px
  margin: 0 auto
  margin-top: -25px
  border: 1px solid #DAD8DE
  position: relative
  background-color: white
  cursor: pointer
  -webkit-box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.15)
  -moz-box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.15)
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.15)

.circle-button > img
  display: block
  margin-top: 17px
  margin-left: auto
  margin-right: auto

.circle-button:hover
    box-shadow: 0px 0px 0px 0px rgba(0,0,0,0.3)

.button-wrapper
  margin-top: 50px

.background
  display: block
  margin: 0 auto


