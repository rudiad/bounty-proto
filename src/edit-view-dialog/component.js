import React, { Component } from 'react';
import PropTypes from 'prop-types';
import closeButton from '../img/close_icon.png';
import './component.sass';

export class EditViewDialog extends Component {
  constructor(args) {
    super(args);
    this.state = {
      x: 0,
      y: 0,
    }
  }

  onChange = (input) => {
    const newState = {};
    newState[input.target.name] = input.target.value;
    this.setState(newState);
  }

  componentDidMount() {
    this.props.views.forEach(element => {
      if (element.id === this.props.idToEdit) {
        this.setState({
          x: element.x,
          y: element.y,
        });
      }
    });
  }

  render() {
    if (!this.props.show) {
      return null;
    }

    return (
      <div className="edit-view">
        <div className="edit-view__background"/>
        <div className="edit-view__dialog">
          <div className="dialog__top-bar-container">
            <div className="top-bar-container__title"> Edit View </div>
            <div className="top-bar-container__escape" onClick={this.props.closeHandler}><img alt="Close" src={closeButton}/></div>
          </div>
          <hr className="dialog__divider"/>
          <div className="edit-view__content">
            <div className="type-and-size-container__type-title">
                  Position (%)
            </div>
              <div className='overlay-custom-container'>
                <div className='overlay-custom-container'>
                <div className="edit-subcontainer__input">
                  <label className="inputs__option-label">X</label>
                  <input type="text" name="x" value={this.state.x} onChange={this.onChange}/>
                </div>
                <div className="edit-subcontainer__input">
                  <label className="inputs__option-label">Y</label>
                  <input type="text" name="y" value={this.state.y} onChange={this.onChange}/>
                </div>
                </div>
              </div>
            </div>
          <hr className="dialog__divider"/>
          <div className="dialog_bottom-bar">
            <div className="bottom-bar__save" onClick={() => this.props.saveViewHandler({ x: this.state.x, y: this.state.y })}> Save </div>
            <div className="bottom-bar__cancel" onClick={this.props.closeHandler}> Cancel </div>
          </div>
        </div>
      </div>
    );
  }
}

EditViewDialog.propTypes = {
  show: PropTypes.bool.isRequired,
  idToEdit: PropTypes.string.isRequired,
  views: PropTypes.array.isRequired,
  closeHandler: PropTypes.func.isRequired,
  saveViewHandler: PropTypes.func.isRequired,
};
