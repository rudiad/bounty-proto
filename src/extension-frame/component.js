import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

const { ExtensionPlatform, ExtensionViewType} = window['extension-coordinator'];

const IFRAME_CLASS = 'extension-frame';
const EXTENSION_FRAME_INIT_ACTION = 'extension-frame-init';

export class ExtensionFrame extends Component {
  componentDidMount() {
    if (this.iframe) {
      this.iframe.onload = this._extensionFrameInit;
    }
  }

  render() {
    return (
      <iframe
      ref={this._bindIframeRef}
        src={process.env.PUBLIC_URL + '/extension-frame.html'}
        frameBorder={0}
        className={'rig-frame ' + IFRAME_CLASS}
        title={this.props.frameId}/>
    );
  }

  _bindIframeRef = (iframe) => {
    this.iframe = iframe;
  }

  _extensionFrameInit = () => {
    const extension = {
      anchor: this.props.type,
      channelId: this.props.extension.channelId,
      loginId: null,
      extension: this.props.extension,
      mode: this.props.mode,
      platform: (this.props.type === ExtensionViewType.Mobile) ? ExtensionPlatform.Mobile : ExtensionPlatform.Web,
      trackingProperties: {},
      iframeClassName: IFRAME_CLASS,
    }
    const data = {
      extension: extension,
      action: EXTENSION_FRAME_INIT_ACTION,
      frameId: this.props.frameId,
    }
    this.iframe.contentWindow.postMessage(data, '*');
  }

  _onIdentityLinked(isLinked) {}

  _onFrameDoubleClick(evt) {
    evt.preventDefault();
  }

  _onModalRequested(confirmationRequest) {}
}

ExtensionFrame.propTypes = {
  className: PropTypes.string.isRequired,
  frameId: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
};
