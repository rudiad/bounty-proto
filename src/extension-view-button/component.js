import React, { Component } from 'react';
import PropTypes from 'prop-types';
import line from '../img/rig_line.png';
import plus from '../img/plus_icon.png';
import './component.sass'

export class ExtensionViewButton extends Component {
  render() {
    return (
      <div className='button-wrapper'>
        <img alt='line' className='background' src={line} />
        <div className="circle-button" onClick={this.props.onClick}>
          <img alt='Add new view' src={plus} />
        </div>
      </div>
    )
  }
}

ExtensionViewButton.propTypes = {
  onClick: PropTypes.func.isRequired
}
