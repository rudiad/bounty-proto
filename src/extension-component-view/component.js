import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ExtensionFrame } from '../extension-frame';
const { ExtensionMode, ExtensionViewType, getComponentPositionFromView, getComponentSizeFromView } = window['extension-coordinator'];

export class ExtensionComponentView extends Component {
  computeViewStyles() {
    const extension = this.props.extension;
    const positionFromView = getComponentPositionFromView(
      this.props.overlaySize.width,
      this.props.overlaySize.height,
      {
        x: this.props.position.x * 100,
        y: this.props.position.y * 100,
      });
    const sizeFromView = getComponentSizeFromView(
      this.props.overlaySize.width,
      this.props.overlaySize.height,
      extension.views.component);

    let viewStyles = {
      border: '1px solid #7D55C7',
      position: 'absolute',
      left: positionFromView.x + 'px',
      top: positionFromView.y + 'px',
      width: `${sizeFromView.width}px`,
      height: `${sizeFromView.height}px`,
    }

    if (extension.views.component.zoom) {
      viewStyles = {
        width: `${sizeFromView.width / sizeFromView.zoomScale}px`,
        height: `${sizeFromView.height / sizeFromView.zoomScale}px`,
        transformOrigin: '0 0',
        transform: `scale(${sizeFromView.zoomScale})`,
      }
    }

    return viewStyles;
  }

  render() {
    return (
      <div
        className="view component-view"
        style={{
          width: this.props.overlaySize.width + 'px',
          height: this.props.overlaySize.height + 'px',
        }}>
          <div style={this.computeViewStyles()}>
          <ExtensionFrame
            className="view"
            frameId={`frameid-${this.props.id}`}
            extension={this.props.extension}
            type={ExtensionViewType.Component}
            mode={ExtensionMode.Viewer}
          />
        </div>
      </div>
    );
  }
}

ExtensionComponentView.propTypes = {
  id: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  overlaySize: PropTypes.object.isRequired,
  position: PropTypes.object.isRequired,
  role: PropTypes.string,
};
