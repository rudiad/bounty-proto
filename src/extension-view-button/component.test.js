import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { ExtensionViewButton } from './component';

describe('<ExtensionViewButton />', () => {
  const setupShallow = setupShallowTest(ExtensionViewButton, () => ({
    onClick: jest.fn(),
  }));

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('fires onClick when clicked on', () => {
    const { wrapper } = setupShallow();
    wrapper.find('.circle-button').simulate('click');
    expect(wrapper.instance().props.onClick).toHaveBeenCalled();
  });
});
