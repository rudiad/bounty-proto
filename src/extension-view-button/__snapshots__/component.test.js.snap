// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionViewButton /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionViewButton
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="line"
          className="background"
          src="rig_line.png"
/>,
        <div
          className="circle-button"
          onClick={[Function]}
>
          <img
                    alt="Add new view"
                    src="plus_icon.png"
          />
</div>,
      ],
      "className": "button-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "line",
          "className": "background",
          "src": "rig_line.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Add new view"
            src="plus_icon.png"
/>,
          "className": "circle-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Add new view",
            "src": "plus_icon.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="line"
            className="background"
            src="rig_line.png"
/>,
          <div
            className="circle-button"
            onClick={[Function]}
>
            <img
                        alt="Add new view"
                        src="plus_icon.png"
            />
</div>,
        ],
        "className": "button-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "line",
            "className": "background",
            "src": "rig_line.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Add new view"
              src="plus_icon.png"
/>,
            "className": "circle-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Add new view",
              "src": "plus_icon.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
